lib/lroPollStrategy.ts

```diff
@@ -335,7 +349,17 @@ class AzureAsyncOperationLROPollStrategy extends LROPollStrategy {
    */
   protected sendPollRequest(): Promise<void> {
     return this.getStatus(this._azureAsyncOperationHeaderValue!).then((response: HttpOperationResponse) => {
+      const statusCode: number = response.status;
       const parsedResponse: any = response.parsedBody;
+      if (statusCode !== 200 && statusCode !== 201 && statusCode !== 202 && statusCode !== 204) {
```

Comment:
Should any of these status code checks ever vary by the expected responses present in the OperationSpec?

Reply:
If we ever get to specifying LRO polling status codes in the OperationSpec/Swagger, then we can revisit this, but I don't believe we do that now.